src/Message.ts

```diff
@@ -134,18 +134,51 @@ export default class DicomMessage {
     var stream = new ReadBufferStream(buffer),
       useSyntax: NormalizedSyntax = EXPLICIT_LITTLE_ENDIAN;
     stream.reset();
+    // Skip the File Preamble (128 bytes)
     stream.increment(128);
+    // Check the DICOM Prefix
     if (stream.readString(4) != "DICM") {
       throw new Error("Invalid: not a dicom file");
     }
-    var el = DicomMessage.readTag(stream, useSyntax),
+    var el = DicomMessage.readTag(stream, useSyntax);
+
+    var metaLength
+
+    // File Meta Information Group Length is correctly suplied
+    if (el.tag.value == 0x00020000) {
       metaLength = el.values[0];
+    // No Length but there are metadata tags
+    } else if (el.tag.value>>16 == 2) {
+      console.warn("Warning: missing File Meta Information Group Length! (0002,0000)");
+
+      // Try to calculate the size of the metadata block by reading all (0002,xxxx) tags
+      while (true) {
```

Comment:
Is `readTag` going to throw an error if you run out of stream?

Otherwise, this doesn't seem to have an unconditional `break` if there's nothing following the metadata section.

Reply:
I mean yeah, I tried not to be super-defensive with "obviously broken files", so those would fail trying to read the stream:
![image](https://github.com/user-attachments/assets/811645c5-9550-4e05-ba89-05d133c4bc8f)

For well-formed (and partially well-formed) files, some tags will follow the metadata section and we'll likely be able to read the file. For non-well-formed files we'll get a read error which will show up in the caller as failed anonymization.